### DISCUSSION TITLE
there is a probability that publishers and subscribers cannot match on the same machine.

### DISCUSSION THREAD

**Discussion by zip-link** [eProsima/Fast-DDS#3182] [Q&A]
I found that when shared memory and udp communication are enabled at the same time, there is a probability that publishers and subscribers cannot match on the same machine. When the problem reappears, I can subscribe to the topic through another device on the LAN. I checked the default simple discovery configuration. After multiple transmissions are enabled at the same time, the PDP will use the 7400 port of the udp broadcast in the same machine, while the EDP uses the shared memory port. I suspect there is a problem with the contents of the shared memory port? How should I position this problem? How to view d in the shared memory port？

**Comment by zip-link** [comment]
My qos configuration is as follows	

                     `participantQos.transport().use_builtin_transports = false; 
    		       auto shm_transport = std::make_shared<SharedMemTransportDescriptor>();
			shm_transport->segment_size(10 * 1024 * 1024);
    		       shm_transport->max_message_size(5 * 1024 * 1024);
			shm_transport->port_queue_capacity(512);
    		        participantQos.transport().user_transports.push_back(shm_transport);
			auto udp_transport = std::make_shared<UDPv4TransportDescriptor>();
			udp_transport->non_blocking_send = true;
			udp_transport->sendBufferSize = 60*1024*1024*8;
			udp_transport->receiveBufferSize = 60*1024*1024*8;
			participantQos.transport().user_transports.push_back(udp_transport);
			participantQos.wire_protocol().builtin.discovery_config.leaseDuration = Duration_t(2, 0);
			participantQos.wire_protocol().builtin.discovery_config.leaseDuration_announcementperiod = Duration_t(1, 0);`

**Comment by Fangel29** [comment]
Hello, has this issue been resolved? I am currently facing the same problem, but I haven't found a solution yet. If you have already resolved it, could you please share how you did it?

**Comment by EugenioCollado** [comment]
Hello,

The transports instantiated in the provided configuration use a similar setup as the default builtin transports in the latest versions of fastdds (both UDP and shared memory (SHM) are enabled by default). That said,  the discovery process (PDP and EDP) is done via UDP even though UDP and SHM are both enabled, so this should be no problem for the discovery of publishers and subscribers and unrelated to the entity's host.

@Fangel29 could you let me know which version of Fast DDS you're using? This will help us determine if any specific version-related issues are affecting your setup.
